### PR TITLE
update doc for Oracle translate function

### PR DESCRIPTION
**Pull request by wzru**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

To add doc for new feature, supporting Oracle translate function

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

https://github.com/pingcap/tidb/pull/26408

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: functions-and-operators/string-functions.md [hunk 1]
```diff
@@ -7,63 +7,65 @@ aliases: ['/docs/dev/functions-and-operators/string-functions/','/docs/dev/refer
...
+| [`SUBSTRING()`](https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_substring)                                             | Return the substring as specified                                                                                                         |
+| [`SUBSTRING_INDEX()`](https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_substring-index)                                 | Return a substring from a string before the specified number of occurrences of the delimiter                                              |
+| [`TO_BASE64()`](https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_to-base64)                                             | Return the argument converted to a base-64 string                                                                                         |
+| [`TRANSLATE()`](https://docs.oracle.com/en/database/oracle/oracle-database/21/sqlrf/TRANSLATE.html#GUID-80F85ACB-092C-4CC7-91F6-B3A585E3A690) | Replace all occurrences of characters by other characters in a string.                                                                    |
```
Comment (kennytm) [inline functions-and-operators/string-functions.md:64]: should we specify the behavior of empty toStr somewhere

**Review by XuHuaiyu — APPROVED** [review]
LGTM

**Comment by TomShawn** [comment]
@kennytm PTAL again, thanks!

**Review by kennytm — APPROVED** [review]

File: functions-and-operators/string-functions.md [hunk 2]
```diff
@@ -6,64 +6,65 @@ aliases: ['/docs/dev/functions-and-operators/string-functions/','/docs/dev/refer
...
+| [`SUBSTRING()`](https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_substring)                                             | Return the substring as specified                                                                                                         |
+| [`SUBSTRING_INDEX()`](https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_substring-index)                                 | Return a substring from a string before the specified number of occurrences of the delimiter                                              |
+| [`TO_BASE64()`](https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_to-base64)                                             | Return the argument converted to a base-64 string                                                                                         |
+| [`TRANSLATE()`](https://docs.oracle.com/en/database/oracle/oracle-database/21/sqlrf/TRANSLATE.html#GUID-80F85ACB-092C-4CC7-91F6-B3A585E3A690) | Replace all occurrences of characters by other characters in a string. It won't treat empty strings as `NULL` as Oracle does              |
```
Comment (TomShawn) [inline functions-and-operators/string-functions.md:63]:
```suggestion
| [`TRANSLATE()`](https://docs.oracle.com/en/database/oracle/oracle-database/21/sqlrf/TRANSLATE.html#GUID-80F85ACB-092C-4CC7-91F6-B3A585E3A690) | Replace all occurrences of characters by other characters in a string. It does not treat empty strings as `NULL` as Oracle does              |
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**